### PR TITLE
In order to ensure that the api documentation is up-to-date in search…

### PR DESCRIPTION
**Pull request by shy-share**
This PR fixes https://github.com/apache/pulsar/pull/20390

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by shy-share** [comment]
@tisonkun cc

**Review by tisonkun — CHANGES_REQUESTED** [review]
1. Why do you delete the sitemap entry?
2. Allow and Disallow seems reversed.

**Comment by shy-share** [comment]
> 1. Why do you delete the sitemap entry?
> 2. Allow and Disallow seems reversed.

I have already edited it

**Comment by shy-share** [comment]
> 1. Why do you delete the sitemap entry?
> 2. Allow and Disallow seems reversed.

@tisonkun 
image

**Review by tisonkun — CHANGES_REQUESTED** [review]
The goal is that we want legacy API docs not to be crawled. Please check the motivation and target,

**Comment by shy-share** [comment]
> The goal is that we want legacy API docs not to be crawled. Please check the motivation and target,

Well, I understand the meaning is just the opposite

**Comment by Anonymitaet** [comment]
> Well, I understand the meaning is just the opposite

We want to show the latest stable API doc version (e.g., 3.0.x) instead of legacy ones.

File: static/robots.txt [hunk 1]
```diff
@@ -1 +1,5 @@
 Sitemap: https://pulsar.apache.org/sitemap.xml
+User-agent: *
+Allow: /
+Disallow: /api/
```
Comment (tisonkun) [inline static/robots.txt:4]: We should exclude legacy versions only. Maintained versions should be still indexed.

**Comment by shy-share** [comment]
#623  has solved this problem, so close this pr

**Pull request closed without merging**